### DISCUSSION TITLE
Unique VTEP IP on a mlag Leaf-Pair

### DISCUSSION THREAD

**Discussion by pradeeprgodhala** [aristanetworks/avd#3975] [Data Model Schemas] [answered]
Hi, 

i am trying a design requirement on Single DC L3LS which require a unique VTEP per L3 Leaf switches which are in MLAG. But i am running across a validation error when i assign a unique vtep_loopback: x.x.x.x to one of the fabric(leaf-a) switches. is this something that is supported from AVD. i think i can use cvp-static-configlets to override AVD generated config.  but want to avoid anything static. 

_l3leaf:
  defaults:
    # Definition of pool of IPs to be used as Virtual Tunnel EndPoint (VXLAN origin and destination IPs)
    **vtep_loopback_ipv4_pool: 192.168.11.0/24**

  node_groups:
    - group: DRT-S200-A01-L3-LEAF
      # ASN to be used by BGP for the group. Both devices in the MLAG pair will use the same BGP ASN
      bgp_as: 65101
      nodes:
        # Definition of hostnames under the node_group
        - name: drt-s200-a01-leaf-a
          id: 1
          mgmt_ip: 10.104.16.82/25
          # Definition of the port to be used in the uplink device facing this device.
          # Note that the number of elements in this list must match the length of 'uplink_switches' as well as 'uplink_interfaces'
          uplink_switch_interfaces:
            - Ethernet1
            - Ethernet1
          **vtep_loopback: '192.168.11.10'**_

Validation error when i run the playbook.

_TASK [arista.avd.eos_designs : Set eos_designs facts] *******************************************************************************************************************

[WARNING]: [drt-s200-a01-leaf-a]: 'Validation Error: l3leaf.node_groups[0].nodes[0].vtep_loopback': '192.168.11.10/32' does not match 'Loopback[\\d/]+'_

when the config is finally generated avd ignores loopback1 and doesnt generate any config.

_interface Loopback0
   description EVPN_Overlay_Peering
   no shutdown
   ip address 192.168.10.3/32
!
interface Loopback101
   description PROD_VTEP_DIAGNOSTICS
   no shutdown
   vrf PROD
   ip address 192.168.13.3/32
!_

Thanks
Pradeep

**Comment by ClausHolbechArista** [comment]
AVD tries to generate best-practice configurations, and having unique VTEP IPs on an MLAG pair is not part of any design guide. We _do_ have support for multi-VTEP when combined with EVPN Multicast, where the MLAG peers have a unique IP (loopback0) _and_ a shared IP (loopback1). Please elaborate on your network design, as I am confused why you need unique IPs.

The `vtep_loopback` variable is allowing you to set the _interface_ which should be the VTEP, so typically `Loopback1` but could be something else as well.

**Reply by pradeeprgodhala** [reply]
we are in the process of migrating our Datacenter deployments models (currently we have a python script written in 2016, and cloud builders 2020), to a single deployment model across two DC's with AVD. Part of the migration, we would have to re-deploy the two DC's to fit the AVD deployment model, which all the IP address have to be changed (including, p2p, vtep ip, evpn diagnostic loopack, mlag p2p, ibgp p2p), we dont have any concern changing the underaly, overlay , mlag ip address, except the vtep(which will be the OLD VTEP) which has to be consistent during the migration. during the migration we want to statically set the vtep IP to the old IP, so we dont disrupt the next hop IP for the evpn type-3 and type-5 routes. 

After the migration is completed, we would then push out the AVD generated VTEP IP to both the switches at the same time in a downtime window. 

we are also exploring the idea of MLAG pair having two different VTEP IP's. would it cause MAC-FLAP in EVPN  or would it cause any disruption, the reason is because during the migration there is a brief period when leaf-a is managed by AVD(from a config prespective, with new IP's p2p, mlag ip, evpn diagnostic ip) and leaf-b has the old config.

Hope this helps. also, we can use static configlets with different vtep IP's. but want to explore the idea of avd doing that. 

Thanks
Pradeep

**Reply by ClausHolbechArista** [reply, marked as answer]
Thank you for the details. I would suggest using `custom_structured_configuration_vxlan_interface` on each device. You can even do this with inline jinja to pick up the temporary VTEP IPs from some file.
